blockchain/state/iterator.go

```diff
@@ -148,14 +153,98 @@ func (it *NodeIterator) retrieve() bool {
 	// Otherwise retrieve the current entry
 	switch {
 	case it.dataIt != nil:
-		it.Hash, it.Parent = it.dataIt.Hash(), it.dataIt.Parent()
+		it.Type = "storage"
+		if it.dataIt.Leaf() {
+			it.Type = "storage_leaf"
+		}
+
+		it.Hash, it.Parent, it.Path = it.dataIt.Hash(), it.dataIt.Parent(), it.dataIt.Path()
+
 		if it.Parent == (common.Hash{}) {
 			it.Parent = it.accountHash
 		}
 	case it.Code != nil:
+		it.Type = "code"
 		it.Hash, it.Parent = it.codeHash, it.accountHash
 	case it.stateIt != nil:
-		it.Hash, it.Parent = it.stateIt.Hash(), it.stateIt.Parent()
+		it.Type = "state"
+		if it.stateIt.Leaf() {
+			it.Type = "state_leaf"
+		}
+
+		it.Hash, it.Parent, it.Path = it.stateIt.Hash(), it.stateIt.Parent(), it.stateIt.Path()
 	}
 	return true
 }
+
+// CheckStateConsistency checks the consistency of all state/storage trie of given two state database.
+func CheckStateConsistency(oldDB database.DBManager, newDB database.DBManager, root common.Hash) error {
```

Comment:
What do you think of using`reflect.DeepEqual` for checking?

Reply:
`reflect.DeepEqual` is better? :)

Reply:
If you are checking all of the variables, it is could be better. It makes sure there is no error in our code.

Reply:
`state.NodeIterator` has pointers which leads to unequality. I will take my suggestion back. Thank you for considering.